testdata/e2e/e2e_helper.go

```diff
@@ -182,13 +215,98 @@ func shouldSkipFileCheck(fileType string, k8sMinorVersion int) bool {
+// (nodes.jsonl) *corev1.Node, and (services.jsonl/replicationcontrollers.jsonl) *LabelMapMatchedResource). The informer
+// data is stored in jsonl format with each line representing one kubernetes object and this function decodes the
+// jsonl line by line.
+// nolint: funlen, gocyclo
+func ByRefFnInformer(refFn k8sRefFn) UnmarshalForK8sListFn {
+	return func(fname string, fdata []byte, parsedK8sList *ParsedK8sLists) error {
+		reader := bytes.NewReader(fdata)
+		dec := json.NewDecoder(reader)
+		// loop reading 1 line at a time until EOF
+		for {
+			// assign storage to the resource list (ex: &p.Pods) that was passed in from the knownFileTypes. And get the
+			// k8s object type from knownFileTypes so the json decoder knows what type of object fields to decode
+			storage, k8sObject := refFn(parsedK8sList)
+			if err := dec.Decode(k8sObject); err != nil {
+				if err == io.EOF {
+					break
+				}
+				return err
+			}
+			// since k8sObject is technically an interface, we need to type switch (thus casting) the k8sObject and
+			// append the object to the correct ParsedK8sLists field (ex: Pods.Items, Deployments.Items,
+			// Namespaces.Items). storage is also technically an interface, so we need to cast this correctly based
+			// on the parsedK8sList field that is passed in (ex: &p.Pods, &p.Services)
+			switch typedK8sObject := k8sObject.(type) {
+			case *corev1.Namespace:
+				typedStorage, ok := storage.(*NamespaceList)
+				if !ok {
+					return errors.New("cannot cast NamespaceList")
+				}
+				typedStorage.Items = append(typedStorage.Items, *typedK8sObject)
+			case *corev1.Pod:
+				typedStorage, ok := storage.(*PodList)
+				if !ok {
+					return errors.New("cannot cast PodList")
+				}
+				typedStorage.Items = append(typedStorage.Items, *typedK8sObject)
+			case *corev1.Node:
+				typeStorage, ok := storage.(*NodeList)
+				if !ok {
+					return errors.New("cannot cast NodeList")
+				}
+				typeStorage.Items = append(typeStorage.Items, *typedK8sObject)
+			case *corev1.PersistentVolume:
+				typeStorage, ok := storage.(*PersistentVolumeList)
+				if !ok {
+					return errors.New("cannot cast PVList")
+				}
+				typeStorage.Items = append(typeStorage.Items, *typedK8sObject)
+			case *corev1.PersistentVolumeClaim:
+				typeStorage, ok := storage.(*PersistentVolumeClaimList)
+				if !ok {
+					return errors.New("cannot cast PVCList")
+				}
+				typeStorage.Items = append(typeStorage.Items, *typedK8sObject)
+			case *LabelSelectorMatchedResource:
+				typeStorage, ok := storage.(*LabelSelectorMatchedResourceList)
+				if !ok {
+					return errors.New("cannot cast LabelSelectorMatchedResourceList")
+				}
+				typeStorage.Items = append(typeStorage.Items, *typedK8sObject)
+			case *LabelMapMatchedResource:
+				typeStorage, ok := storage.(*LabelMapMatchedResourceList)
+				if !ok {
+					return errors.New("cannot cast LabelMapMatchedResourceList")
+				}
+				typeStorage.Items = append(typeStorage.Items, *typedK8sObject)
+			}
```

Comment:
Default case with an error returned could be useful here.

Reply:
👍 When I made changes in cake-services I forgot to update this. I also pulled the refFn call out of the loop here as well. TY